src/System.Private.CoreLib/src/System/Diagnostics/Eventing/RuntimeEventSource.cs

```diff
@@ -0,0 +1,143 @@
+    internal sealed class RuntimeEventSource : EventSource
+    {
+        // Process-level EventCounters
+        private EventCounter _totalProcessTimeCounter;
+        private EventCounter _workingSetCounter;
+        private EventCounter _virtualMemorySizeCounter;
+        private EventCounter _handleCountCounter;
+        private EventCounter _threadCountCounter;
+
+        // GC Counters
+        private EventCounter _gcTotalMemoryCounter;
+        private EventCounter _gcGen0CollectionCounter;
+        private EventCounter _gcGen1CollectionCounter;
+        private EventCounter _gcGen2CollectionCounter;
+
+        // Exception
+        private EventCounter _exceptionCounter;
+
+        private Timer m_timer;
+
+        private const int EnabledPollingIntervalMilliseconds = 1000; // 1 second
+
+        // Threads
+        // TODO
+
+        
+        public RuntimeEventSource(): base(EventSourceSettings.EtwSelfDescribingEventFormat)
+        {
+
+        }
+
+        protected override void OnEventCommand(System.Diagnostics.Tracing.EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // TODO: These are all returning fake stuff now
+                _totalProcessTimeCounter = new EventCounter("Total Process Time", this);
+                _workingSetCounter = new EventCounter("Working Set", this);
+                _virtualMemorySizeCounter = new EventCounter("Virtual Memory Size", this);
+                _handleCountCounter = new EventCounter("Handle Count", this);
+                _threadCountCounter = new EventCounter("Thread Count", this);
+
+                // GC counters
+                _gcTotalMemoryCounter = new EventCounter("Total Memory by GC", this);
+                _gcGen0CollectionCounter = new EventCounter("Gen 0 GC Count", this);
+                _gcGen1CollectionCounter = new EventCounter("Gen 1 GC Count", this);
+                _gcGen2CollectionCounter = new EventCounter("Gen 2 GC Count", this);
+
+                // TODO: Expose a managed API for computing this
+                _exceptionCounter = new EventCounter("Exception Count", this);
+                
+
+                // Initialize the timer, but don't set it to run.
+                // The timer will be set to run each time PollForTracingCommand is called.
+                m_timer = new Timer(
+                    callback: new TimerCallback(PollForCounterUpdate),
+                    state: null,
+                    dueTime: Timeout.Infinite,
+                    period: Timeout.Infinite,
+                    flowExecutionContext: false);
+
+                // Trigger the first poll operation on the start-up path.
+                PollForCounterUpdate(null);
+            }
+            else if (command.Command == EventCommand.Disable)
+            {
+                // Dispose counters when perfcounters are disabled
+
+                _totalProcessTimeCounter.Dispose();
```

Comment:
I would add comments or make a note somewhere that there are multi-threading issues that need to be resolved. These objects are being manipulated from both the OnEventCommand thread and also from a timer callback thread and some synchronization is necessary. Exactly what synchronization will depend on the thread-safety guarantees EventCounter can provide and the final API for working with it, which I think we are waiting on @vancem for.